ublox_gps/src/gps.cpp

```diff
@@ -116,6 +116,17 @@ void Gps::initializeSerial(std::string port, unsigned int baudrate,
   }
 
   ROS_INFO("U-Blox: Opened serial port %s", port.c_str());
+  {
+    // Wait for at least one char on serial port before returning from read.
+    // Sadly Boost doesn't provide any way to set this.
+    int fd = serial->native_handle();
+    struct termios tio;
+    tcgetattr(fd, &tio);
+    tio.c_cc[VTIME] = 0;
+    tio.c_cc[VMIN] = 1;
+    tcflush(fd, TCIFLUSH);
+    tcsetattr(fd, TCSANOW, &tio);
+  }
```

Comment:
This block is not required since we set the correct mode for the serial port in the code just below this.

Reply:
the code right below this looks like it checks the boost version before setting the correct mode for the serial port. this code block above looks like it is meant to wait for at least one character from the serial port before returning from read....

Have you tested it without this block above to make sure that it is not needed? With different versions of boost?

Reply:
It looks like this block of code is from this comment: https://github.com/KumarRobotics/ublox/issues/20#issuecomment-321747684, so I'm fairly confident about it.
The Boost version check was added since Boost sets the correct parameters when initializing the serial port in versions >= 1.66.